### PR TITLE
feat(sql): add regr_r2, regr_rmse, and regr_tstats aggregate functions

### LINKED ISSUE #6320: add `regr_r2` group by function, and funcs for RMSE and t-stats
### Is your feature request related to a problem?

We have a series of stats functions, for example `regr_slope`, `stddev...` etc.

W are missing a few useful ones to cover:

- rsquared values
- root mean square error
- t-statistics

Just group by aggregates for now, unless we find a strong need for window functions too.

### Describe the solution you'd like.

_No response_

### Describe alternatives you've considered.

_No response_

### Full Name:

Nick Woolmer

### Affiliation:

QuestDB

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ritoban23**
## Add regr_r2, regr_rmse, and regr_tstats aggregate functions 

This PR implements three new SQL aggregate functions for regression analysis in QuestDB:

- **`regr_r2(y, x)`**: Coefficient of determination (R²)
- **`regr_rmse(y, x)`**: Root Mean Square Error (RMSE)
- **`regr_tstats(y, x)`**: t-statistic for regression slope significance

Fixes #6320 

### Features

- Numerically stable single-pass computation (Welford’s algorithm)
- Handles all edge cases: nulls, zero variance, perfect fits, overflow
- Parallel aggregation support via `merge()`
- Compatible with PostgreSQL’s `regr_*` semantics

### Testing

- **Total tests:** 40 (13 R², 13 RMSE, 14 t-stats)
- **Status:** All tests passing
- Covers: perfect fits, null handling, type conversions, overflow, edge cases

### Files Added

- `core/src/main/java/io/questdb/griffin/engine/functions/groupby/RegressionR2FunctionFactory.java`
- `core/src/main/java/io/questdb/griffin/engine/functions/groupby/RegressionRmseFunctionFactory.java`
- `core/src/main/java/io/questdb/griffin/engine/functions/groupby/RegressionTStatsFunctionFactory.java`
- `core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/RegressionR2FunctionFactoryTest.java`
- `core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/RegressionRmseFunctionFactoryTest.java`
- `core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/RegressionTStatsFunctionFactoryTest.java`

### CODE REVIEW

**Comment by bluestreak01** [comment]
@coderabbit review. Check for implementation completeness, check for test coverage, check for thread-safety and that multi-threaded tests are present. Also check if there are performance issues, java allocations, or redundant execution on critical path.

File: core/src/main/java/io/questdb/griffin/engine/functions/groupby/RegressionR2FunctionFactory.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+    @Override
+    public boolean isGroupBy() {
+        return true;
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) {
+        return new RegressionR2Function(args.getQuick(0), args.getQuick(1));
+    }
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/engine/functions/groupby/RegressionR2FunctionFactory.java:57]: @Coderabbit function registration is optional, it is only requires for ambiguous functions. Otherwise function loader scans the classpath. Don't worry about registrations anymore
